src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -224,7 +228,21 @@ protected override int ExecuteTool(string pathToTool, string responseFileCommand
             // '/out' parameter has to have an existing directory.
             Directory.CreateDirectory(Path.GetDirectoryName(_outputR2RImage));
 
+            WarningsDetected = false;
+
             return base.ExecuteTool(pathToTool, responseFileCommands, commandLineCommands);
         }
+
+        protected override void LogEventsFromTextOutput(string singleLine, MessageImportance messageImportance)
+        {
+            if (!ShowCompilerWarnings && singleLine.IndexOf("warning:", StringComparison.OrdinalIgnoreCase) != -1)
```

Comment:
@rainersigwald Should `!= -1` rather be changed to `== 0`?

Reply:
I would probably say "yes" but I don't know anything about the actual format of crossgen errors.

Reply:
What it was doing previously was fed through

https://github.com/microsoft/msbuild/blob/ee12fc2666de515d1981822e404ea40a8c5f71d5/src/Shared/CanonicalError.cs#L65-L81

Which admits some things but not everything before literal `warning:`. If crossgen uses any of that, I'd leave it as is; if it only does `^warning:` I'd scan for just that.